File: client_cli.py
from socket_client import Client
from os import system
import json

with open("config.json", "r") as file:
    data = json.load(file)


def print_intro():
    print(f"""
         111   1   1    1    11111
        11     11111   1 1     1
        11  1  1   1  11111    1
         111   1   1  1   1    1\n
    name: {data["config"]["name"]}
    ip: {data["config"]["ip"]}
    port: {data["config"]["port"]}\n
    type 'help' for get manual
       """)


def write_in_conf(param, value):
    data["config"][param] = value
    with open("config.json", "w") as file:
        json.dump(data, file)


def handle_commands(command):
    match command:
        case "set_name":
            name = input("Enter your name: ")
            write_in_conf("name", name)
            system('clear')
            print_intro()
        case "set_ip":
            ip = input("Enter server ip: ")
            write_in_conf("ip", ip)
            system('clear')
            print_intro()
        case "set_port":
            port = input("Enter server port: ")
            write_in_conf("port", port)
            system('clear')
            print_intro()
        case "set_key":
            key = input("Enter your key: ")
            write_in_conf("key", key)
            system('clear')
            print_intro()
        case "clear":
            system('clear')
        case "exit":
            exit(0)
        case "help":
            with open("help", 'r') as file:
                documentation = file.readlines()
                print(*documentation)
        case "connect":
            if Client.name:
                Client.connect_to_chat()
            else:
                print("You didn't set your name")
        case _:
            print("command not found")


if __name__ == "__main__":
    print_intro()
    try:
        while True:
            command = input("> ").strip()
            handle_commands(command)
    except KeyboardInterrupt:
        print("Quiting")
        exit(0)

File: client_cli.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <map>
#include "json.hpp"
using namespace std;


void print_intro();
void handle_commands(string command);


map <string, string> json = json_to_map("config.json");


int main(int argc, char *argv[])
{
	print_intro();
	string command;
	while (true)
	{
		cout << "> ";
		getline(cin, command);
		handle_commands(command);
	}
	return 0;
}


void print_intro()
{
	cout << "\n\t 111   1   1    1    11111" << endl
		<< "\t11     11111   1 1     1" << endl
		<< "\t11  1  1   1  11111    1" << endl
		<< "\t 111   1   1  1   1    1\n" << endl
		<< "\t\tname: " << json["name"] << endl
		<< "\t\tip: " << json["ip"] << endl
		<< "\t\tport: " << json["port"] << endl
		<< "\t\tkey: " << json["key"] << endl
		<< "\n\ttype 'help' for get manual\n" << endl;
}


void reload()
{
	system("clear");
	map_to_json("config.json", json);
	print_intro();
}


void handle_commands(string command)
{
	if (command == "set_name")
	{
		cout << "Enter your name > ";
		cin >> json["name"];
		reload();
	}
	else if (command == "set_key")
	{
		cout << "Enter your key > ";
		cin >> json["key"];
		reload();
	}
	else if (command == "set_ip")
	{
		cout << "Enter server ip > ";
		cin >> json["ip"];
		reload();
	}
	else if (command == "set_port")
	{
		cout << "Enter server port > ";
		cin >> json["port"];
		reload();
	}
	else if (command == "help")
	{
		ifstream file ("help");
		char mychar;
		while ( file )
		{
			mychar = file.get();
			cout << mychar;
		}
		file.close();
	}
	else if (command == "connect") { system("python3 socket_client.py"); }
	else if (command == "exit") { exit(0); }
	else if (command == "clear") { reload(); }
	else if (command == "") { }
	else { cout << "Non valid command" << endl; }
}
